precompiles/dapps-staking/DappsStaking.sol

```diff
@@ -29,6 +29,12 @@ interface DappsStaking {
     /// @return amount, Staked amount by the staker
     function read_staked_amount(bytes calldata staker) external view returns (uint128);
 
+    /// @notice Read Staked amount on a given contract for the staker
+    /// @param contract_id contract evm address
+    /// @param staker in form of 20 or 32 hex bytes
+    /// @return amount, Staked amount by the staker
+    function read_staked_amount_on_contract(address contract_id, bytes calldata staker) external view returns (uint128);
```

Comment:
This won't be WASM compatible.

But it also shouldn't be a problem since no EVM contract can query WASM related stuff yet since there's nothing to really query.